File: CPProject/food.cpp
#include "food.hpp"

Food::Food(int id, double price, int volume, Supplier supplier) 
      : Product(id, price, volume, supplier){}

double Food::calculate_discount(int quantity, int month)
{
      return Product::calculate_discount(quantity, month);
}

Food::~Food(){}



//* --------------------------------Food concrete subclasses-------------------------------- *//
Milk::Milk(int id, double price, int volume, Supplier supplier)
        : Food(id, price, volume, supplier){}

std::string Milk::display_product_info() const
{
      std::string info = "[Milk: ";
      info += "ID: " + std::to_string(getId()) + ", ";
      info += "Supplier: " + getSupplier().getName() + ", ";
      info += "Price: " + std::to_string(getPrice()) + "]";
      return info;
}

double Milk::calculate_discount(int quantity, int month)
{
      return Food::calculate_discount(quantity, month);
}

Milk::~Milk(){}


Cheese::Cheese(int id, double price, int volume, Supplier supplier)
        : Food(id, price, volume, supplier){}

std::string Cheese::display_product_info() const
{
      std::string info = "Cheese: ";
      info += "ID: " + std::to_string(getId()) + ", ";
      info += "Supplier: " + getSupplier().getName() + ", ";
      info += "Price: " + std::to_string(getPrice()) + "]";
      return info;
}

double Cheese::calculate_discount(int quantity, int month)
{
      return Food::calculate_discount(quantity, month);
}

Cheese::~Cheese(){}

File: application/src/main/java/project/product/Food.java
package project.product;

import project.stock.StockRepository;
import project.supplier.Supplier;

public abstract class Food extends Product{

    public Food(Supplier supplierId, int volume, double price) {
        super(supplierId, volume, price);
    }

    @Override
    public String display_product_info() {
        return "Food{" +
                ", productId=" + getProductId() +
                ", supplierId=" + getSupplier().getIdSupplier() +
                ", supplier name=" + getSupplier().getName() +
                ", stock=" + StockRepository.getStockQuantity(this) +
                ", price=" + getPrice() + 
                '}';
    }
}
